ragna/ui/components/file_uploader.py

```diff
@@ -0,0 +1,274 @@
+import json
+from random import random
+
+import param
+from panel.reactive import ReactiveHTML
+from panel.widgets import Widget
+
+
+class FileUploader(ReactiveHTML, Widget):  # type: ignore[misc]
+    allowed_documents = param.List(default=[])
+    allowed_documents_str = param.String(default="")
+
+    file_list = param.List(default=[])
+
+    custom_js = param.String(default="")
+    uploaded_documents_json = param.String(default="")
+
+    title = param.String(default="")
+
+    def __init__(self, allowed_documents, token, informations_endpoint, **params):
+        super().__init__(**params)
+
+        self.token = token
+        self.informations_endpoint = informations_endpoint
+
+        self.after_upload_callback = None
+
+    def can_proceed_to_upload(self):
+        return len(self.file_list) > 0
+
+    @param.depends("allowed_documents", watch=True)
+    def update_allowed_documents_str(self):
+        if len(self.allowed_documents) == 1:
+            self.allowed_documents_str = (
+                "Only " + self.allowed_documents[0] + " files are allowed."
+            )
+        else:
+            self.allowed_documents_str = "Allowed files : " + ", ".join(
+                self.allowed_documents
+            )
+
+    @param.depends("uploaded_documents_json", watch=True)
+    def did_finish_upload(self):
+        if self.after_upload_callback is not None:
+            self.after_upload_callback(json.loads(self.uploaded_documents_json))
+
+    def perform_upload(self, event=None, after_upload_callback=None):
+        self.after_upload_callback = after_upload_callback
+
+        self.loading = True
+
+        final_callback_js = """
+            var final_callback = function(uploaded_documents) {
+                self.get_uploaded_documents_json().innerText = JSON.stringify(uploaded_documents);
+            };
+        """
+
+        # This is a hack to force the re-execution of the javascript.
+        # If the whole javascript is the same, and doesn't change,
+        # the panel widget is not re-renderer, and the upload function is not called.
+        random_id = f" var random_id = '{str(random())}';"
```

Comment:
This gives a random float. Should we use a different scheme like I suggested above with a random UUID?

Reply:
it totally doesn't matter here, but I'll switch to uuid as well.